### ISSUE TITLE
what is the reason the loss fluctuates largely?

### ISSUE THREAD

**Issue by cheer37** [BVLC/caffe#3801]
My training loss fluctuates largely. but it decreases overall.
So i tried to make a low learning rate, but the same result.
Training displaying step is not the small either.
What do you think about it?
Any opinion will be appreciated.

**Comment by iamprem** [comment]
![Is this how it looks like for you?](http://cs231n.github.io/assets/nn3/loss.jpeg)

If this is how it looks like for you, you may consider increasing the batch size as suggested [here](http://cs231n.github.io/neural-networks-3/#loss)

I'm a beginner, so disregard if this is irrelevant for you.
Thanks

**Comment by cheer37** [comment]
Thanks iamprem.
But I can't compare my training curve with this curve, because they were applied to different application so that has different scale.
I compared the my curve with others 's curve applied to same field thus same net structure, same dataset, but not only learning parameters.
By the way, how can batch size affect the learning curve fluctuation? would you be more detail?
And also, i want to know the relation between batch size and accuracy, either.
Thanks.

**Comment by iamprem** [comment]
Quoting from [here](http://cs231n.github.io/neural-networks-3/#loss)
> The amount of "wiggle" in the loss is related to the batch size. When the batch size is 1, the wiggle will be relatively high. When the batch size is the full dataset, the wiggle will be minimal because every gradient update should be improving the loss function monotonically (unless the learning rate is set too high).

Please read through that modules to get clear understanding. Also i think the scale on y-axis shouldn't matter.

**Comment by cheer37** [comment]
Thank you for you nice information.

**Comment by limTeTE** [comment]
@cheer37 This is because the nature of mini-batch based gradient descent. if you see the overall trend is decreasing and in some iteration the loss just increased, that's fine, because you just happened to met some bad batches which cannot learn much from it. In fact, the oscalliating trend decreases from SGD, mini-batch based GD to GD in which your mini-batch is just the whole training set and you should see the loss function is strictly decreasing in each epoch. This is what i learnt from Andres Ng's deep learning lecture. Hope this would somehow helps you if it's still not too late.

**Comment by Keramatfar** [comment]
From SGD, we know that in each iteration some weights must get better, but how this improvement, reduce accuracy of test?
